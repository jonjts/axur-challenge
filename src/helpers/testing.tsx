import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import ThemeProvider from '../providers/Theme';
import InspectionProvider from '../providers/Inspection'

export const putProviders = (component: JSX.Element) => (
    <ThemeProvider>
        <InspectionProvider>
            {component}
        </InspectionProvider>
    </ThemeProvider>
);

export const matchSnapshotWithProviders = (component: JSX.Element) => {
    const tree = renderer.create(putProviders(component)).toJSON();
    expect(tree).toMatchSnapshot();
    return tree;
};

export const renderWithProviders = (component: JSX.Element) => {
    return render(putProviders(component));
};
