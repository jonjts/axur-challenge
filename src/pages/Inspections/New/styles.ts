import styled from 'styled-components';

export const FormGroup = styled.div`
    display: flex;
    flex-direction: column;

    input {
        @media (min-width: 0px) {
            width: 100%;
        }

        @media (min-width: 992px) {
            width: 50%;
        }
    }

    :focus {

    }
`;

export const Footer = styled.div`
    min-height: 80px;
    display: flex;
    flex: 1;
    flex-direction: row;
    justify-content: flex-end;
    align-items: center;
    margin-top: ${({ theme: { spacings: { xl } } }) => xl.pixel};
    border-top: 1px solid ${({ theme: { colors } }) => colors.default};

    a {
        margin-right: ${({ theme: { spacings: { lg } } }) => lg.pixel};
    }
`;