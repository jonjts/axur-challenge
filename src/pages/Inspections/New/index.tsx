import React, { useState, useCallback, useEffect } from 'react';
import { useHistory } from 'react-router-dom';

import { useInspection } from '../../../providers/Inspection';

import Card from '../../../components/Card';

import { FormGroup } from './styles';

const New: React.FC = () => {
  const { goBack } = useHistory();
  const { insert, loading } = useInspection();

  const [keyword, setKeyword] = useState('');
  const [hasError, setHasError] = useState(false);

  useEffect(() => {
    setHasError(false);
  }, [keyword]);

  const handleSave = useCallback(async () => {
    try {
      await insert(keyword);
      goBack();
    } catch (error) {
      setHasError(true);
    }
  }, [insert, keyword, goBack]);


  return (
    <>
      <h1>Nova Inspeção</h1>
      <Card
        showBackButton
        showSubmitButton
        backText='Voltar'
        submitText='Salvar'
        onBackClicked={goBack}
        onSubmitClicked={handleSave}
        disableButtons={loading}
      >
        <FormGroup hasError={hasError}>
          <label>Keyword</label>
          <input
            placeholder='Informe a keyword'
            value={keyword}
            onChange={({ target: { value } }) => setKeyword(value)}
          />
        </FormGroup>
      </Card>
    </>
  );
};

export default New;