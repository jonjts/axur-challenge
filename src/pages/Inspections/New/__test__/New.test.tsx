import { matchSnapshotWithProviders, renderWithProviders } from '../../../../helpers/testing';

import New from '../index';

describe('New Inspection', () => {
  test('match with snapshot', () => {
    matchSnapshotWithProviders(<New />);
  });

  test('renders correctly', () => {
    renderWithProviders(<New />);
  });
});
