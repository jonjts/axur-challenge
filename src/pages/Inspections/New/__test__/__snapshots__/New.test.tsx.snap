// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Inspection match with snapshot 1`] = `
Array [
  <h1>
    Nova Inspeção
  </h1>,
  <div
    className="sc-hKgILt RVPmz"
  >
    <div
      className="sc-jSgupP hBDZzH"
    >
      <label>
        Keyword
      </label>
      <input
        onChange={[Function]}
        placeholder="Informe a keyword"
        value=""
      />
    </div>
    <div
      className="sc-eCssSg cqYcHT"
    >
      <button
        className="sc-dlfnbm kacgkK"
        data-testid="btn-link"
      >
        Voltar
      </button>
      <button
        className="sc-gsTCUz dPeWPU"
        data-testid="btn"
        onClick={[Function]}
      >
        Salvar
      </button>
    </div>
  </div>,
]
`;
